app/src/main/cpp/SystemUtils.h

```diff
@@ -7,13 +7,19 @@
 namespace crow {
 
 #define ANDROID_OS_BUILD_ID "ro.build.id"
+#define ANDROID_OS_MODEL_ID "ro.product.vendor.model"
 
 // Get the Build ID of the current Android system.
 inline const char* GetBuildIdString(char* out) {
   __system_property_get(ANDROID_OS_BUILD_ID, out);
   return out;
 }
 
+// Retrieves the model name from Android OS. Returns the amount of chars of the model.
+inline int PopulateDeviceModelString(char* model) {
+  return __system_property_get(ANDROID_OS_MODEL_ID, model);
```

Comment:
what if model pointer is null ?

Reply:
It's a precondition of the method, do you suggest to add an assert?

Reply:
I just tried that: it crashes.

Reply:
> It's a precondition of the method, do you suggest to add an assert?

Not sure, being an inline method. But perhaps we should have some checks before calling the function.

Reply:
Just to isolate the calls to android APIs in a different file

Reply:
In any case the calls to that method are controller by us, it isn't that depend on some random variable.